src/tasks.js

```diff
@@ -89,6 +89,13 @@ ko.tasks = (function () {
             }
         },
 
+        isScheduled: function(func) {
+            var keys = Object.keys(taskQueue);
+            for (var a = keys.length - 1; a >= 0; a--)
+                if (taskQueue[keys[a]] === func) return true;
```

Comment:
How n^2?  It's single pass over the queue so that would make scheduling a task go from O(1) to O(n).

It would be somewhat mitigated by the performance improvement of avoiding recomputing computed observables needlessly.  I don't know how often this happens in practice though (other than in the circular dependency edge case)

Another idea would be to use `Set` for the taskQueue which automatically eliminates duplicates.  It also conveniently returns entries in insertion order so easily facilities a queue.  But that's a larger re-factor.  Also Sets are not available on all browsers that Knockout is currently supporting (though to mitigate there are Set [polyfills](https://github.com/WebReflection/es6-collections) )